prismarine-viewer/viewer/lib/entities.ts

```diff
@@ -454,6 +464,7 @@ export class Entities extends EventEmitter {
     // ---
     // not player
     const displayText = entity.metadata?.[3] && this.parseEntityLabel(entity.metadata[2])
+      || entity.metadata?.[23] && this.parseEntityLabel(entity.metadata[23]) // text displays
```

Comment:
Is the key (index number) consistent across the versions that support it? We can check it entities data of minecraft-data module

Reply:
Also, it is definitely not right here! The `3` index is `custom_name_visible` property for entities. However 23 for text_display is text and for block_display is `block_state` property (not text), so it will result in error spamming in the console. So we should do a check here, however I might think of a better solution. Definitely code needs to be refactored (I'll gladly help you here). At least we should use property names, not number :d

Reply:
sadly dont have permissions to push

Reply:
> sadly dont have permissions to push

Sorry, forgot that GitHub treats org forks differently from personal ones :S Will make PRs from a personal one in the future.



> Also, it is definitely not right here! The `3` index is `custom_name_visible` property for entities. However 23 for text_display is text and for block_display is `block_state` property (not text), so it will result in error spamming in the console. So we should do a check here, however I might think of a better solution. Definitely code needs to be refactored (I'll gladly help you here). At least we should use property names, not number :d

You are right, I didn't consider that. I kinda thought the indexes are entity-specific but it makes sense that they aren't. Feel free to leave PRs open in the future for me to fix such stuff directly in them. Will open another PR which improves the situation here (and maybe even adds more features of the text displays already, will have to look at how difficult to do that is with the current name tag system)